### DISCUSSION TITLE
How to pass data between view models?

### DISCUSSION THREAD

**Discussion by D-Diyare** [lepoco/wpfui#484] [Q&A]
Today I wanted to give this beautiful library a shot, so I was playing around with it and everything was just fine until I tried to test some stuff as I usually do in MVVM pattern which one of them is passing data between view models, I went to the navigation documentation but didn't found anything related to this case, the idea is for example: ProductListViewModel we have many products and each of them has details page, I'm trying to pass the product Id to the product details view model.
how to work with this scenario, any workaround for that?
Thanks in advance

**Comment by xaberue** [comment]
I have the same problem; can anyone show some basic sample about how to do it?

Many thanks!

**Comment by xaberue** [comment]
Not sure if it's the best approach, but here's what I've done and it works for me:

1. Don't register the associated ViewModel of the target Page in the DI container, since we will create it from the page invoking it.
2. Create the target ViewModel from the root page, and inject the WPF UI INavigationService in the page that will invoke the redirection.
3. Use the method that accepts a Page type (registered in the DI container) and the view model to it. The service will automatically assign it to the Page's DataContext.

```csharp

var viewModel = new YourViewModel(); //Add all the required data for your VM here
_navigationService.Navigate(typeof(YourPage), viewModel);

```

That's all, 

I'll appreciate if anyone has a better alternative and wants to share it here as well.

Regards

**Reply by lpsky** [reply]
```
you  can use DI,
var viewModel = App.GetService<YourViewModel>();
```
